include/zephyr/bluetooth/gatt.h

```diff
@@ -1994,10 +1994,11 @@ struct bt_gatt_read_params {
  *  the callback being called with @p data set to @c NULL.
  *
  *  Note that returning @ref BT_GATT_ITER_CONTINUE is really starting a new ATT
- *  operation, so this can fail to allocate resources. However, all API errors
- *  are reported as if the server returned @ref BT_ATT_ERR_UNLIKELY. There is no
- *  way to distinguish between this condition and a @ref BT_ATT_ERR_UNLIKELY
- *  response from the server itself.
+ *  operation, so this can fail to allocate resources. However, if the received
+ *  data length is invalid, the error @ref BT_ATT_ERR_INVALID_PDU will be
+ *  returned. For all other API errors are reported as if the server returned
+ *  @ref BT_ATT_ERR_UNLIKELY. There is no way to distinguish between this
+ *  condition and a @ref BT_ATT_ERR_UNLIKELY response from the server itself.
```

Comment:
Putting the new documentation here in between the sentences is not right, since this change has nothing to do with the callback returning `BT_GATT_ITER_CONTINUE`. Let's separate it out to its own paragraph to maintain clarity.

We might also instead put the documentation on `bt_gatt_read_func_t`.